src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkTask.java

```diff
@@ -46,6 +49,12 @@ public void start(final Map<String, String> props) {
     config = new JdbcSinkConfig(props);
     initWriter();
     remainingRetries = config.maxRetries;
+    try {
+      reporter = context.errantRecordReporter();
+    } catch (NoSuchMethodError | NoClassDefFoundError e) {
+      // Will occur in Connect runtimes earlier than 2.6
+      reporter = null;
```

Comment:
WDYT about providing our own errant-record-reporter implementation that formats, logs, and re-throws the exception in this case, and restores the prior behavior?
I think it would simplify/remove all the `reporter == null` checks later and simplify the logic.

If anything, seeing a more specific error from a smaller batch as the final error may be more informative, and if the smaller batches actually prevent errors, the error reporter won't be called at all!

Reply:
Can you elaborate what you mean? Maybe you could give an example?

Reply:
Something like:

```suggestion
      reporter = (sinkRecord, throwable) -> {
        throw new ConnectException("Encountered error when writing sink record", throwable);
      };
```
It's a little messy as you can't throw the throwable directly, since the method signature doesn't have `throws Throwable` in it, but it still does the job.

Reply:
Or really, since `reporter` can be null coming out of the `try`, you'd have to do this check outside of the catch, and replace both the pre-2.6 behavior, and the error-reporter-not-configured behavior with this lambda.

Reply:
Nevermind, this requires a definition of ErrantRecordReporter, which won't be available on <2.6.